rust/arrow/src/compute/kernels/take.rs

```diff
@@ -576,186 +768,255 @@ mod tests {
         _test_take_string::<LargeStringArray>()
     }
 
-    #[test]
-    fn test_take_list() {
-        // Construct a value array, [[0,0,0], [-1,-2,-1], [2,3]]
-        let value_data = Int32Array::from(vec![0, 0, 0, -1, -2, -1, 2, 3]).data();
-        // Construct offsets
-        let value_offsets = Buffer::from(&[0, 3, 6, 8].to_byte_slice());
-        // Construct a list array from the above two
-        let list_data_type = DataType::List(Box::new(DataType::Int32));
-        let list_data = ArrayData::builder(list_data_type.clone())
-            .len(3)
-            .add_buffer(value_offsets)
-            .add_child_data(value_data)
-            .build();
-        let list_array = Arc::new(ListArray::from(list_data)) as ArrayRef;
+    macro_rules! test_take_list {
```

Comment:
The diff isn't formatting too nicely here, but the changes convert the existing `List` tests into macro versions with the same names, then each macro is used twice, once for regular `List`, and once for `LargeList`. 

* `test_take_list`
* `test_take_list_with_value_nulls`
* `test_take_list_with_list_nulls`